admin_class.php

```diff
@@ -1,690 +1,1172 @@
+    $stmt_check = $this->db->prepare("SELECT * FROM {$type[$_SESSION['login_type']]} WHERE email = ? AND id != ?");
+    $stmt_check->bind_param("si", $email, $id);
+    $stmt_check->execute();
+    $result_check = $stmt_check->get_result();
+
+    if($result_check->num_rows > 0){
+        $stmt_check->close();
+        return 2; // Email already exists
+    }
+
+    $types = str_repeat("s", count($_POST) - 4); // Assuming all parameters are strings except id, cpass, table, and password
+
+    if(isset($_FILES['img']) && $_FILES['img']['tmp_name'] != ''){
+        $fname = strtotime(date('y-m-d H:i')).'_'.$_FILES['img']['name'];
+        $move = move_uploaded_file($_FILES['img']['tmp_name'],'assets/uploads/'. $fname);
+        $data .= ", avatar = ? ";
+        $types .= "s";
+    }
+
+    if(!empty($password))
+        $data .= " ,password = md5(?) ";
+
+    if(empty($id)){
+        $stmt = $this->db->prepare("INSERT INTO {$type[$_SESSION['login_type']]} SET $data");
+    }else{
+        $data .= " WHERE id = ? ";
+        $types .= "i"; // Assuming id is an integer
+        $stmt = $this->db->prepare("UPDATE {$type[$_SESSION['login_type']]} SET $data");
+    }
+
+    // Bind parameters
+    $params = array_values($_POST);
+    if(isset($_FILES['img']) && $_FILES['img']['tmp_name'] != ''){
+        $params[] = $fname;
+    }
+    if(!empty($id)){
+        $params[] = $id;
+    }
+    $stmt->bind_param($types, ...$params);
+
+    // Execute query
+    $save = $stmt->execute();
+    $stmt->close();
+
+    if($save){
+        foreach ($_POST as $key => $value) {
+            if($key != 'password' && !is_numeric($key))
+                $_SESSION['login_'.$key] = $value;
+        }
+        if(isset($_FILES['img']) && !empty($_FILES['img']['tmp_name']))
+                $_SESSION['login_avatar'] = $fname;
+        return 1;
+    }
+}
+
+function delete_user(){
+    extract($_POST);
+    
+    // Prepare the query
+    $stmt = $this->db->prepare("DELETE FROM users WHERE id = ?");
+    $stmt->bind_param("i", $id);
+    $delete = $stmt->execute();
+    $stmt->close();
+
+    if($delete)
+        return 1;
+}
+
+function save_system_settings(){
+    extract($_POST);
+    $data = '';
+    
+    // Construct the update query
+    foreach($_POST as $k => $v){
+        if(!is_numeric($k)){
+            if(empty($data)){
+                $data .= " $k = ? ";
+            }else{
+                $data .= ", $k = ? ";
+            }
+        }
+    }
+
+    // Check if system_settings table has data
+    $chk = $this->db->query("SELECT * FROM system_settings");
+    if($chk->num_rows > 0){
+        $data_types = str_repeat("s", count($_POST)); // Assuming all parameters are strings
+        $data_values = array_values($_POST);
+        
+        // Prepare the update query
+        $stmt = $this->db->prepare("UPDATE system_settings SET $data WHERE id = ?");
+        $stmt->bind_param($data_types."i", ...$data_values, $chk->fetch_array()['id']);
```

Comment:
Fatal error: Cannot use positional argument after argument unpacking in /Applications/MAMP/htdocs/Faculty_Evaluation_System-main/admin_class.php on line 354
Getting this error

Reply:
Ran it on my local and got this error - How are you testing your changes?